### PR TITLE
Changes to support CoreNEURON as a simulator

### PR DESCRIPTION
**Pull request by pramodk**
* make mod files threadsafe
   - changing necessary GLOBAL variables to RANGE
   - change state_discontinuity to direct assignments
     as per Michael's suggestion
 * Add coreneuron config parameters in run.py and simConfig.py
 * Change random numbers to Random123 if CoreNEURON is enabled
 * Use newer copy of vecstim.mod from NEURON that is now compatible
   with NEURON as well as CoreNEURON

Co-authored-by: Ioannis Magkanaris <iomagkanaris@gmail.com>

### CODE REVIEW

File: examples/M1detailed/mod/gabab.mod [hunk 1]
```diff
@@ -196,14 +197,14 @@ NET_RECEIVE(weight,  r0, t0 (ms)) {
 		r0 = weight*(Rinf + (r0 - Rinf)*exp(-(t - t0)/Rtau))
 		t0 = t
 		synon = synon - weight
-		state_discontinuity(Ron, Ron - r0)
```
Comment (iomaganaris) [inline examples/M1detailed/mod/gabab.mod:199]: https://neurondev.slack.com/archives/CS8R6MU2Y/p1616004374008800

**Comment by iomaganaris** [comment]
I have tested the `M1detailed` circuit with the latest `NEURON` and `CoreNEURON` on CPU and GPU and the spikes are nearly identical between the two. The differences in the spikes between them lies in some spikes shifted in time. I have tried `online` and `offline` modes for transferring data from `NEURON` to `CoreNEURON` as well as debug builds and I noticed the same behavior.

File: examples/M1detailed/mod/kapin.mod [hunk 2]
```diff
@@ -50,14 +50,15 @@ ASSIGNED {       :parameters needed to solve DE
...
+    qt
 }
 
 
 STATE {          :the unknown parameters to be solved in the DEs 
 	n l
 }
 
-LOCAL qt
+:LOCAL qt
```
Comment (pramodk) [inline examples/M1detailed/mod/kapin.mod:61]: `nitpick`: just delete this

File: examples/M1detailed/mod/kapcb.mod [hunk 3]
```diff
@@ -47,13 +47,14 @@ ASSIGNED {       :parameters needed to solve DE
...
 	ki		(mM)
+    qt
 }
 
 STATE {          :the unknown parameters to be solved in the DEs 
 	n l
 }
 
-LOCAL qt
+:LOCAL qt
```
Comment (pramodk) [inline examples/M1detailed/mod/kapcb.mod:57]: nitpick: just delete this

**Review by pramodk — COMMENTED** [review]
Overall all changes look good. 
(For the differences, need some time to look into if this is expected. We can do this ~next week.)

**Pull request merged**